lib/src/connector/connector.cc

```diff
@@ -305,6 +305,13 @@ bool Connector::isAssociated() const
     return isConnected() && session_association_.success.load();
 }
 
+ConnectionTimings Connector::getConnectionTimings() const
+{
+    return (connection_ptr_ == nullptr
+                ? ConnectionTimings()
```

Comment:
Temporary storage depends on context, if I'm correct. More here: http://www.learncpp.com/cpp-tutorial/79-the-stack-and-the-heap/